compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -160,9 +160,34 @@ public BallerinaSemanticModel(BLangPackage bLangPackage, CompilerContext context
 
     // Private helper methods for the public APIs above.
 
-    private boolean isSymbolInUserProject(BSymbol symbol, DiagnosticPos cursorPos) {
+    private boolean isSymbolInUserProject(BSymbol symbol, Location cursorPos) {
+        int value = 0;
+
+        LinePosition cursorPosStartLine = cursorPos.lineRange().startLine();
+        LinePosition symbolStartLine = symbol.pos.lineRange().startLine();
+
+        if (bLangPackage.packageID.equals(symbol.pkgID)) {
+            // If the package detail strings are same, then compare the start line.
+            if (cursorPosStartLine.line() < symbolStartLine.line()) {
+                value = -1;
+            } else if (cursorPosStartLine.line() > symbolStartLine.line()) {
+                value = 1;
+            }
+
+            if (value == 0) {
+                // If the start line is the same, then compare the start column.
+                if (cursorPosStartLine.offset() < symbolStartLine.offset()) {
+                    value = -1;
+                } else if (cursorPosStartLine.offset() > symbolStartLine.offset()) {
+                    value = 1;
+                }
+            }
```

Comment:
Can simplify this as:
```java
if (symbol.owner.getKind() == SymbolKind.PACKAGE || Symbols.isFlagOn(symbol.flags, Flags.WORKER) {
   return true;
}

if (cursorPosStartLine.line() < symbolStartLine.line()) {
    return false;
} 

if (cursorPosStartLine.line() > symbolStartLine.line()) {
    return true;
}

// means line numbers are equal. check for the column numbers
return cursorPosStartLine.offset() <= symbolStartLine.offset();

Reply:
btw, shouldn't we check for the in-range kinda check?
Check here for a similar one: https://github.com/ballerina-platform/ballerina-lang/blob/master/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java#L32

Reply:
Yeah I think the name of the function should be changed

Reply:
The current one only seems to check for the start line/columns info only.

If we are doing a range check, shouldn't we check for the ending line/column info as well?

Reply:
what this function does is checking the current cursor position has passed the position of symbol. So do we need to check the `endline`